File: 6.08/서지훈_백준_26091_현대모비스_소프트웨어_아카데미.cpp
#include <iostream>
#include <algorithm>

int main(){
	int N, M;
	std::cin >> N >> M;
	int *arr = new int[N];
	for(int i = 0; i < N; ++i){
		std::cin >> arr[i];
	}
	
	std::sort(arr, arr+N);
	
	int left = 0, right = N-1;
	
	int result = 0;
	while(left < right){
		if(arr[left] + arr[right] >= M){
			result += 1;
			right -= 1;
		}
		left += 1;
	}
	std::cout << result;
	
	delete arr;
}

File: 6.08/서지훈_백준_26091_현대모비스_소프트웨어_아카데미.py
def main():
    N, M = map(int, input().split())
    li = sorted(list(map(int, input().split())))
    left = 0
    right = N-1

    result = 0
    while left < right:
        if li[left] + li[right] >= M:
            result += 1
            right -= 1
        left += 1
    print(result)


main()